include/mbedtls/gcm.h

```diff
@@ -41,66 +49,69 @@ extern "C" {
 #endif
 
 /**
- * \brief          GCM context structure
+ * \brief          The GCM context structure.
  */
 typedef struct {
-    mbedtls_cipher_context_t cipher_ctx;/*!< cipher context used */
-    uint64_t HL[16];            /*!< Precalculated HTable */
-    uint64_t HH[16];            /*!< Precalculated HTable */
-    uint64_t len;               /*!< Total data length */
-    uint64_t add_len;           /*!< Total add length */
-    unsigned char base_ectr[16];/*!< First ECTR for tag */
-    unsigned char y[16];        /*!< Y working value */
-    unsigned char buf[16];      /*!< buf working value */
-    int mode;                   /*!< Encrypt or Decrypt */
+    mbedtls_cipher_context_t cipher_ctx;  /*!< The cipher context used. */
+    uint64_t HL[16];                      /*!< Precalculated HTable low. */
+    uint64_t HH[16];                      /*!< Precalculated HTable high. */
+    uint64_t len;                         /*!< The total length of the data. */
+    uint64_t add_len;                     /*!< The total length of the additional data. */
+    unsigned char base_ectr[16];          /*!< The first ECTR for tag. */
+    unsigned char y[16];                  /*!< The Y working value. */
+    unsigned char buf[16];                /*!< The buf working value. */
+    int mode;                             /*!< The operation to perform: 
+                                               #MBEDTLS_GCM_ENCRYPT or 
+                                               #MBEDTLS_GCM_DECRYPT. */
 }
 mbedtls_gcm_context;
 
 /**
- * \brief           Initialize GCM context (just makes references valid)
- *                  Makes the context ready for mbedtls_gcm_setkey() or
- *                  mbedtls_gcm_free().
+ * \brief           This function initializes the specified GCM context, 
+ *                  to make references valid, and prepares the context 
+ *                  for mbedtls_gcm_setkey() or mbedtls_gcm_free().
  *
- * \param ctx       GCM context to initialize
+ * \param ctx       The GCM context to initialize.
  */
 void mbedtls_gcm_init( mbedtls_gcm_context *ctx );
 
 /**
- * \brief           GCM initialization (encryption)
+ * \brief           This function initializes the specified GCM context.
```

Comment:
Blocker, pre-existing defect: This has to be distinguished from the documentation of `mbedtls_gcm_init`. 
Suggestion: Let's be uniform with the generic MD layer, by adding `This function doesn't bind the GCM context to a particular cipher, nor set the key. For this purpose, use mbedtls_gcm_setkey()` to the documentation of `mbedtls_gcm_init` above, and by using `This function sets up the GCM context for encryption and decryption by binding it to a block cipher type and setting the keys to use.`

Reply:
Done

Reply:
As I mentioned elsewhere IIRC: Mbed TLS generally uses “setup” for this phase. Thus: “This function sets up the specified GCM context.” And “GCM context to set up” below. Although, here, it might be better to use a more precise description: “Set a cipher algorithm and key for a GCM context.” or “Associate a GCM context with a cipher algorithm and a key.”.